applications/plugins/Compliant/Compliant_test/Assembly_test.cpp

```diff
@@ -133,13 +133,18 @@ struct Assembly_test : public CompliantSolver_test
         ConstantForceField3::SPtr ff = New<ConstantForceField3>();
         string1->addObject(ff);
         helper::vector<unsigned>* indices = ff->d_indices.beginEdit(); // not managed to create a WriteAccessor with a resize function for a ConstantForceField::SetIndex
-        helper::WriteAccessor< Data<helper::vector<Vec3> > > forces( ff->d_forces );
-        (*indices).resize(2);
-        forces.resize(2);
-        // pull the left-hand particle to the left
-        (*indices)[0]= 0; forces[0]= Vec3(-1,0,0);
-        // pull the right-hand particle to the right
-        (*indices)[1]= n-1; forces[1]= Vec3(1,0,0);
+
+        /// It is mandatory to keep this scope otherwise the WriteAccessor will hold a
+        /// pointer to the ff that will be destroyed.
+        {
+            helper::WriteAccessor< Data<helper::vector<Vec3> > > forces( ff->d_forces );
```

Comment:
I don't understand, if this WriteAccessor is not used later in the code (after the scope you just added) when/why would he access the point he is holding?

Reply:
I think the destructor is called at the end of global scope and if the accessor is referencing an object that has already been detroyed..then the troubles are there.